src/charm.py

```diff
@@ -1,103 +1,292 @@
+            return
+
+        self.unit.status = MaintenanceStatus("Configuring resources")
+        self._container.add_layer(self._container_name, self._pebble_layer, combine=True)
+
+        if not self.model.relations[self._db_relation_name]:
+            self.unit.status = BlockedStatus("Missing required relation with postgresql")
+            event.defer()
+            return
+
+        if not self.database.is_resource_created():
+            self.unit.status = WaitingStatus("Waiting for database creation")
+            event.defer()
+            return
+
+        self._container.push(self._config_file_path, self._render_conf_file(), make_dirs=True)
+        try:
+            self._container.restart(self._container_name)
+        except ChangeError as err:
+            logger.error(str(err))
+            self.unit.status = BlockedStatus("Failed to restart, please consult the logs")
+            return
+
+        self.unit.status = ActiveStatus()
+
+    def _on_pebble_ready(self, event: PebbleReadyEvent) -> None:
+        """Event Handler for pebble ready event."""
+        # Necessary directory for log forwarding
+        if not self._container.can_connect():
+            event.defer()
+            self.unit.status = WaitingStatus("Waiting to connect to glauth container")
+            return
+        if not self._container.isdir(str(self._log_dir)):
+            self._container.make_dir(path=str(self._log_dir), make_parents=True)
+            logger.info(f"Created directory {self._log_dir}")
+
+        self._handle_status_update_config(event)
+
+    def _on_config_changed(self, event: ConfigChangedEvent) -> None:
+        """Event Handler for config changed event."""
+        self._handle_status_update_config(event)
+
+    def _on_database_created(self, event: DatabaseCreatedEvent) -> None:
+        """Event Handler for database created event."""
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to GLAuth container. Deferring event.")
+            self.unit.status = WaitingStatus("Waiting to connect to glauth container")
+            return
+
+        self.unit.status = MaintenanceStatus(
+            "Configuring container and resources for database connection"
+        )
+
+        try:
+            self._container.get_service(self._container_name)
+        except (ModelError, RuntimeError):
+            event.defer()
+            self.unit.status = WaitingStatus("Waiting for glauth service")
+            logger.info("GLAuth service is absent. Deferring database created event.")
+            return
+
+        logger.info("Updating GLAuth config and restarting service")
+        self._container.add_layer(self._container_name, self._pebble_layer, combine=True)
+        self._container.push(self._config_file_path, self._render_conf_file(), make_dirs=True)
+
+        self._container.start(self._container_name)
+        self.unit.status = ActiveStatus()
```

Comment:
Why is this needed?

Reply:
when the db is first connected, I use the container's start method. On other db changes and config changes I use the restart method. Not sure what happens when calling restart method on unstarted container. Let me know if you rather I invoked the _handle_status_update_config as response to DataBaseCreatedEvent.